### DISCUSSION TITLE
How to set the username and password to download private feed packages?

### DISCUSSION THREAD

**Discussion by ghost** [renovatebot/renovate#10743] [Request Help]
I followed the instructions for Azure Devops and not matter how I configure it, I get this error:

`"message": "Response code 401 (Unauthorized)",
         "stack": "HTTPError: Response code 401 (Unauthorized)\n    at Request.<anonymous> (/home/vsts/.npm/_npx/1544/lib/node_modules/renovate/node_modules/got/dist/source/as-promise/index.js:117:42)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
         "options": {
           "headers": {
             "user-agent": "RenovateBot/25.52.5 (https://github.com/renovatebot/renovate)",
             "authorization": "***********",
             "accept": "application/json",
             "accept-encoding": "gzip, deflate, br"
           },
           "url": "https://pkgs.dev.azure.com/[org_name]/_packaging/[project_name]/nuget/v3/index.json",
           "username": "",
           "password": "",
           "method": "GET",
           "http2": false
         },
`

**Comment by viceice** [comment]
https://docs.renovatebot.com/private-modules/

**Comment by ghost** [comment]
My config.js file looks like this and I still get that issue. Is my configuration wrong?

```
module.exports = {
  platform: 'azure',
  logLevel: 'debug',
  endpoint: 'https://dev.azure.com/[my-organization-name]/',
  token: process.env.TOKEN,
  hostRules: [
    {
      hostType: 'npm',
      matchHost: 'pkgs.dev.azure.com',
      username: 'apikey',
      password: process.env.TOKEN,
    },
    {
      hostType: 'azure',
      matchHost: 'https://pkgs.dev.azure.com',
      username: '[my azure email]',
      password: '[hard-coded-PAT-token]',
    },
  ],
  repositories: ['[Project-Name]/[Repository-Name]'],
};
```

With these settings, I still get the error:

`nuget registry failure: can't get RegistrationsBaseUrl (repository=[Project-Name]/[Repository-Name])`

And it would seem the username and password are not being set.

**Reply by viceice** [reply]
i think for nuget you need this host rule:

```js
    {
      matchHost: 'pkgs.dev.azure.com',
      username: 'apikey',
      password: process.env.TOKEN,
    },
```

So whithout `hostType` it will apply to all requests to azure devops.

**Reply by ghost** [reply]
thanks
